src/model.cpp

```diff
@@ -0,0 +1,250 @@
+#include <model.hpp>
+
+namespace nest {
+namespace mc {
+
+// Helper function for building cell groups used by model constructor
+// See bottom of file for definition.
+cell_group_ptr make_cell_group(
```

Comment:
It is a factory for `cell_groups` (that should actually be in the name).
So it can't be part of the `cell_group` interface.
Logically, you can argue  that it belongs in its own file.
I implemented it in `model.cpp` because the constructor of `model` is the only spot where it is used.